src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -129,12 +129,15 @@ SymbolKind.NetModule or
                     CompilationTrackerState state,
                     CompilationAndGeneratorDriverTranslationAction? translate)
                 {
-                    var baseCompilation = state.CompilationWithoutGeneratedDocuments;
-                    if (baseCompilation != null)
+                    var compilationWithoutGeneratedDocuments = state.CompilationWithoutGeneratedDocuments;
+                    if (compilationWithoutGeneratedDocuments != null)
                     {
                         var intermediateProjects = state is InProgressState inProgressState
                             ? inProgressState.IntermediateProjects
                             : [];
+                        var finalCompilationWithGeneratedDocuments = state is FinalState finalState
+                            ? finalState.FinalCompilationWithGeneratedDocuments
+                            : null;
```

Comment:
Do we not prefer this form?

```suggestion
                        var finalCompilationWithGeneratedDocuments = (state as FinalState)?.FinalCompilationWithGeneratedDocuments;
```

Reply:
i started with that, but i preferred just the clarity of testing, and doing somethign when it passed, versus inferring that from processing null.

Reply:
works for me